back-end/Service.py

```diff
@@ -0,0 +1,280 @@
+import os
+from flask import Flask, current_app, flash, jsonify, make_response, redirect, request, url_for, send_from_directory
+from werkzeug.utils import secure_filename
+from flask_cors import CORS
+from DataAccess import *
+from DataAccessCalendar import *
+from DataAccessImage import *
+from DataAccessOrders import *
+from DataAccessLocation import *
+from Image import *
+from ImageHandler import *
+from TimeUnit import *
+from TimeUnitHandler import *
+from LocationHandler import *
+from Location import *
+from Order import *
+from OrderHandler import * 
+from ImageLinkerHandler import *
+from ImageLinker import *
+from Mail import *
+from ConfigProvider import *
+import json
+import FileManipulations
+app = Flask(__name__)
+app.debug = True
+
+CORS(app)
+
+
+app.config ['UPLOAD_FOLDER'] = ConfigProvider.UPLOAD_FOLDER
+
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ConfigProvider.ALLOWED_EXTENSIONS
+
+@app.route('/saveImageFile', methods=['POST'])
+def upload_file():
+    if 'file' not in request.files:
+        flash('No file part')
+        return redirect(request.url)
+    file = request.files['file']
+    if file.filename == '':
+        flash('No selected file')
+        return redirect(request.url)
+    if file and allowed_file(file.filename):
+        file.save(os.path.join(app.config['UPLOAD_FOLDER'], request.form['name']))
+        resp = make_response(jsonify("Saved", 200))
+        return resp
+
+@app.route('/getImageFile/<fileName>', methods=['GET'])
+def get_image_file(fileName):
+    return send_from_directory(UPLOAD_FOLDER, fileName,as_attachment=False)

```

Comment:
send_file is the function that handles sending files to the user. It does no sanity check on the input, so it would also happily send protected/../../../etc/passwd or whatever. In your specific case that might not work, but if you are unaware of that type of attack you might produce unsecure code.

send_from_directory checks wether the requested file is really from the specified directory. That way the above attack would not work.